scb-engine/src/main/java/io/securecodebox/engine/service/AuthService.java

```diff
@@ -0,0 +1,97 @@
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ * /
+ */
+package io.securecodebox.engine.service;
+
+import io.securecodebox.engine.model.PermissionType;
+import io.securecodebox.engine.model.ResourceType;
+import org.camunda.bpm.engine.identity.Group;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Component;
+import org.camunda.bpm.engine.ProcessEngine;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.security.authentication.InsufficientAuthenticationException;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Component
+public class AuthService {
+    private static String AUTH_DISABLED_TYPE = "none";
+    private static final Logger LOG = LoggerFactory.getLogger(AuthService.class);
+
+    @Autowired
+    ProcessEngine engine;
+
+    @Value("${securecodebox.rest.auth}")
+    private String authType;
+
+    public void isAuthorizedFor(String resourceId, ResourceType resource, PermissionType permission) throws InsufficientAuthenticationException{
+        if(AUTH_DISABLED_TYPE.equals(authType)){
+            return;
+        }
+
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+
+        if(authentication == null) {
+            throw new InsufficientAuthenticationException("No authentication provided.");
+        }
+
+        List<String> groups = engine
+                .getIdentityService()
+                .createGroupQuery()
+                .groupMember(authentication.getName())
+                .list()
+                .stream()
+                .map(Group::getId)
+                .collect(Collectors.toList());
+
+        boolean isAuthorized = false;
+        if(resourceId == null){
+            isAuthorized = engine.getAuthorizationService().isUserAuthorized(
+                    authentication.getName(),
+                    groups,
+                    permission.getCamundaPermission(),
+                    resource.getCamundaResource()
+            );
+        } else {
+            isAuthorized = engine.getAuthorizationService().isUserAuthorized(
+                    authentication.getName(),
+                    groups,
+                    permission.getCamundaPermission(),
+                    resource.getCamundaResource(),
+                    resourceId
+            );
+        }
+
+        LOG.debug("Current User '{}' with groups: '{}'", authentication.getName(), groups);
+        LOG.debug("Access check for [{}, {}, {}]: {}", resourceId, resource, permission, isAuthorized);
+
+        if(!isAuthorized){
+            throw new InsufficientAuthenticationException("User is not authorised to perform this action.");
```

Comment:
Technically this is not correct. At this point, the user is authenticated (but not authorized)